lib/memory_cache.dart

```diff
@@ -0,0 +1,27 @@
+import 'dart:collection';
+
+/// Simple class to cache values on the cache
+///
+class MemoryCache<K, V> {
```

Comment:
should we refactor our file and audio caches to use this somehow?

Reply:
I don’t think so, at least not without changing this class a little, on audio and image cache, we don’t automatically clear the cache, everything is manual there.

We could change this class to have an infinity cache and then use it there. But I wouldn’t do this on this PR so we don’t mix stuff

Reply:
yeah totally, I meant as a followup. def. not a part of this PR